core/src/main/scripts/importer/cbioportal_common.py

```diff
@@ -1000,3 +1010,110 @@ def run_java(*args):
+
+    # For backward compatibility, build connection URL from individual properties
+    if properties.get(PORTAL_PROPERTY_DATABASE_URL) is None:
+        if properties.get(PORTAL_PROPERTY_DATABASE_HOST) is None or properties.get(PORTAL_PROPERTY_DATABASE_NAME) is None:
+            print("No connection information provided for database connection. Please set 'db.host' and "
+                "'db.portal_db_name' in portal.properties. Or better, use the 'db.connection_string' "
+                "(see https://docs.cbioportal.org/deployment/customization/portal.properties-reference/)",
+                file=ERROR_FILE)
+            return None
+        print("\n" + "-"*112)
+        print("-- Deprecation warning:")
+        print("-- You are connection to the database using the deprecated 'db.host', 'db.portal_db_name' and "
+            "'db.use_ssl' properties.")
+        print("-- Please use the 'db.connection_string' instead "
+            "(see https://docs.cbioportal.org/deployment/customization/portal.properties-reference/).")
+        print("-"*112 + "\n")
+        properties[PORTAL_PROPERTY_DATABASE_URL] = f"jdbc:mysql://{properties[PORTAL_PROPERTY_DATABASE_HOST]}" \
+            f"/{properties[PORTAL_PROPERTY_DATABASE_NAME]}" \
+            f"?zeroDateTimeBehavior=convertToNull&useSSL={properties[PORTAL_PROPERTY_DATABASE_USESSL]}"
+
+    return PortalProperties(properties[PORTAL_PROPERTY_DATABASE_USER],
+                            properties[PORTAL_PROPERTY_DATABASE_PW],
+                            properties[PORTAL_PROPERTY_DATABASE_URL])
+
+
+def parse_properties_file(properties_filename: str) -> Dict[str, str]:
+
+    if not os.path.exists(properties_filename):
+        print('properties file %s cannot be found' % properties_filename, file=ERROR_FILE)
+        sys.exit(2)
+        
+    properties = {}
+    with open(properties_filename, 'r') as properties_file:
+        for line in properties_file:
+            line = line.strip()
+            # skip line if its blank or a comment
+            if len(line) == 0 or line.startswith('#'):
+                continue
+            try:
+                name, value = line.split('=', maxsplit=1)
+            except ValueError:
+                print(
+                    'Skipping invalid entry in property file: %s' % line,
+                    file=ERROR_FILE)
+                continue
+            properties[name] = value.strip()
+    return properties
+
+
+def get_db_cursor(portal_properties: PortalProperties):
+
+    try:
+        url_elements = dsnparse.parse(portal_properties.database_url)
+        connection_kwargs = {
+            "host": url_elements.host,
+            "port": url_elements.port if url_elements.port is not None else 3306,
+            "db": url_elements.paths[0],
+            "user": portal_properties.database_user,
+            "passwd": portal_properties.database_pw,
+            "ssl": "useSSL" not in url_elements.query or url_elements.query["useSSL"] == "true"
```

Comment:
@pvannierop I tested the local migration script with the new URL parameter, but I got this error: 
```
  File "/Users/zhaog/Desktop/Github/cbioportal/core/src/main/scripts/importer/cbioportal_common.py", line 1108, in get_db_cursor
    "ssl": "useSSL" not in url_elements.query or url_elements.query["useSSL"] == "true"
```
Maybe it should be `url_elements.useSSL == "true"`?

Reply:
But I still get an error about this connection string:
`db.connection_string=jdbc:mysql://localhost:3306/cbioportal?zeroDateTimeBehavior=convertToNull&useSSL=false`
This is the result I get from the output (I hid the username and password)
```
{'host': None, 'port': 3306, 'db': 'mysql:', 'ssl': False}
```
I don't know how could this happen, could you take a look?

Reply:
And db has been parsed as `mysql` seems wrong. Not sure if we can find a better library to do the parsing task?